### PR TITLE
Issue #4833 - Milestone 2

### LINKED ISSUE #4833: [2D MVP] Milestone 2
### Description
Base issue for milestone 2
dependencies:
- [x] https://github.com/3drepo/3drepo.io/issues/4828
- [x] https://github.com/3drepo/3drepo.io/issues/4827
- [x] https://github.com/3drepo/3drepo.io/issues/4829
- [x] https://github.com/3drepo/3drepo.io/issues/4948
- [x] https://github.com/3drepo/3drepo.io/issues/4832
- [x] https://github.com/3drepo/3drepo.io/issues/4987
- [x] https://github.com/3drepo/3drepo.io/issues/4959
- [x] https://github.com/3drepo/3drepo.io/issues/5101
- [x] https://github.com/3drepo/3drepo.io/issues/4937
- [x] https://github.com/3drepo/3drepo.io/issues/5066
- [x] https://github.com/3drepo/3drepo.io/issues/5124
- [x] https://github.com/3drepo/3drepo.io/issues/5006

### Goals
- [ ] As a user I want to see a list of drawings available to view when viewing a federation/container
- [ ] As a user I want the ability to choose a drawing to open and view it side by side with the 3D Model
- [ ] As a user I want the ability to navigate around the 2D drawing
- [ ] As a user I want the ability to create a Screen shot in 2D view.

### Related Resources
[Mockup](https://www.figma.com/file/uTWnaYZ8IyoTn2i2xKtdBo/2D---3D?type=design&node-id=11-9027&mode=design&t=MZd3EAnPzKS7vGBv-0)

### PR DESCRIPTION
**Pull request by sanmont3drepo**
This fixes #4833 

#### Description
- The user can view the drawings in the viewer from the drawings list
- The user can view the drawing and pan and zoom 

### bugs
- [x] [The formatting on the uncalibrated button seems off](https://github.com/3drepo/3drepo.io/pull/5154#discussion_r1752230880) (@Amantini1997 fixed in milestone 3)
- [x] [Set the SVG drawing to have a white background](https://github.com/3drepo/3drepo.io/pull/5154#discussion_r1752250263) (@sanmont3drepo )
- [x] [Background colour of the viewer should change as depicted by viewer settings](https://github.com/3drepo/3drepo.io/pull/5154#discussion_r1752251066) (@The-Daniel)
- [x] [The drawing panel should be visible if the enters the viewer with a drawing ID in the query](https://github.com/3drepo/3drepo.io/pull/5154#discussion_r1752255047) (@Amantini1997 fixed in milestone 3)
- [x] [ I'm looking at a corner of the picture and I can't zoom out to see the whole picture](https://github.com/3drepo/3drepo.io/pull/5154#discussion_r1752272185) ( @sanmont3drepo )
- [x] [Overflow in drawing items in drawing card ](https://github.com/3drepo/3drepo.io/pull/5154#discussion_r1752293735) (@Amantini1997)
- [x] [New drawings are ignored in drawings card](https://github.com/3drepo/3drepo.io/pull/5154#discussion_r1752310173) (@The-Daniel)
- [x] [Scrolling should keep the selected drawing in view (just like custom tickets)
](https://github.com/3drepo/3drepo.io/pull/5154#discussion_r1757258394)
- [x] [Sort the drawings by last updated](https://github.com/3drepo/3drepo.io/pull/5154#discussion_r1757265811)

### CODE REVIEW

File: backend/src/v5/services/eventsListener/components/modelEvents.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ const revisionAdded = async ({ teamspace, project, model, revId, modelType, cali
 			{ _id: 0, tag: 1, author: 1, timestamp: 1, desc: 1, rFile: 1, format: 1, statusCode: 1, revCode: 1 });
```
Comment (carmenfan) [inline backend/src/v5/services/eventsListener/components/modelEvents.js:88]: image
Can we set the SVG drawing to have a white background so it looks like a 2D sheet and also it won't be hindered by users setting a custom background colour

Comment (carmenfan) [inline backend/src/v5/services/eventsListener/components/modelEvents.js:88]: Background colour of the viewer should change as depicted by viewer settings
image

Comment (carmenfan) [inline backend/src/v5/services/eventsListener/components/modelEvents.js:88]: I recall talking to @Amantini1997  about this so maybe it's fixed in a different milestone, but the drawing panel should be visible if the enters the viewer with a drawing ID in the query. I.e. it should look like this:

image

At the moment it looks like this:
image

Comment (carmenfan) [inline backend/src/v5/services/eventsListener/components/modelEvents.js:88]: I think we need to allow the user to zoom out for more than the original image size.

For e.g. here, I'm looking at a corner of the picture and I can't zoom out to see the whole picture without re-centering the image, which doesn't feel natural:

**Review by carmenfan — CHANGES_REQUESTED** [review]

File: backend/src/v5/services/eventsListener/components/modelEvents.js (hunk 1 above)
Comment (carmenfan) [inline backend/src/v5/services/eventsListener/components/modelEvents.js:88]: image
We're missing an overflow strategy here, @apeitup  can you clarify what we should do here?

Comment (carmenfan) [inline backend/src/v5/services/eventsListener/components/modelEvents.js:88]: Drawing list is catching the chat signal for new revisions on existing drawings ok, but new drawings are ignored

**Review by carmenfan — CHANGES_REQUESTED** [review]

File: backend/src/v5/services/eventsListener/components/modelEvents.js (hunk 1 above)
Comment (carmenfan) [reply backend/src/v5/services/eventsListener/components/modelEvents.js:88]: To clarify, the end result should have the area highlighted in red being white (approx) and the area outside of the red boundary should be the current background colour

File: backend/src/v5/services/eventsListener/components/modelEvents.js
Comment (carmenfan) [inline backend/src/v5/services/eventsListener/components/modelEvents.js:1]: Scrolling should keep the selected drawing in view (just like custom tickets)

File: backend/src/v5/services/eventsListener/components/modelEvents.js (hunk 1 above)
Comment (carmenfan) [reply backend/src/v5/services/eventsListener/components/modelEvents.js:88]: we went for ellipsis with tool tip here... can we confirm with @apeitup that this is what we want?
image


@AsiteZane showed me a mockup of multiline instead when I mentioned it
image

File: backend/src/v5/services/eventsListener/components/modelEvents.js
Comment (carmenfan) [inline backend/src/v5/services/eventsListener/components/modelEvents.js:1]: It doesn't look like there's a sort logic on the drawing list at all?

Waiting on @AsiteZane  to confirm by what we should be sorting by 😆

**Review by carmenfan — CHANGES_REQUESTED** [review]

File: backend/src/v5/services/eventsListener/components/modelEvents.js
Comment (AsiteZane) [reply backend/src/v5/services/eventsListener/components/modelEvents.js:1]: Last updated please :)

**Review by carmenfan — APPROVED** [review]

**Pull request merged**